gradio/components/dataset.py

```diff
@@ -61,16 +63,19 @@ def __init__(
             container: If True, will place the component in a container - providing some extra padding around the border.
             scale: relative width compared to adjacent Components in a Row. For example, if Component A has scale=2, and Component B has scale=1, A will be twice as wide as B. Should be an integer.
             min_width: minimum pixel width, will wrap if not sufficient screen space to satisfy this value. If a certain scale value results in this Component being narrower than min_width, the min_width parameter will be respected first.
+            component_props: This is created when the dataset is initialized. Not intended to be set manually. But we add it to init so that Datasets can be loaded from spaces with gr.load.
+            component_ids: This is created when the dataset is initialized. Not intended to be set manually. But we add it to init so that Datasets can be loaded from spaces with gr.load.
         """
-        super().__init__(visible=visible, elem_id=elem_id, elem_classes=elem_classes)
+        super().__init__(visible=visible, elem_id=elem_id, elem_classes=elem_classes, root_url=root_url,
```

Comment:
root_url and _skip_init_processing was not being set for dataset

Reply:
Let's add `render` here as well, its not being passed up right now